src/nvme/ioctl.c

```diff
@@ -355,6 +355,130 @@ int nvme_get_log_page(int fd, __u32 xfer_len, struct nvme_get_log_args *args)
 	return 0;
 }
 
+static int read_ana_chunk(int fd, enum nvme_log_ana_lsp lsp, bool rae,
+			  __u8 *log, __u8 **read, __u8 *to_read, __u8 *log_end)
+{
+	if (to_read > log_end) {
+		errno = ENOSPC;
+		return -1;
+	}
+
+	while (*read < to_read) {
+		__u32 len = min(log_end - *read, NVME_LOG_PAGE_PDU_SIZE);
+		int ret = nvme_get_log_ana(fd, lsp, rae, *read - log, len, *read);
```

Comment:
please have the `nvme_get_log_ana` call not mixed into the variable deceleration block.